src/Services/AbstractResourceService.php

```diff
@@ -176,23 +167,20 @@ public function search(array $payload, array $options = [])
      *
      * @param string $method
      * @param string $url
-     * @param array $options
+     * @param ResourceServiceOptions $options
      *
-     * @return \LiveIntent\Resource|array
+     * @return \LiveIntent\Resource|ResourceResponse
      */
-    public function request(string $method, string $url, array $options = [])
+    public function requestWithOptions(string $method, string $url, ResourceServiceOptions $options = null)
     {
-        $response = $this->requestRaw($method, $url, $options);
-
-        if (array_key_exists('keepRawResponse', $options)) {
-            return [
-                'resource' => $this->newResource($response->json()['output']),
-                'headers' => $response->headers(),
-                'body' => $response->body(),
-            ];
+        $response = $this->requestRaw($method, $url);
```

Comment:
It looks like this throws an exception when it's not a 2XX response. I imagine we'll want to catch this and _not_ throw it when `$options->keepRawResponse` is `true`. Let's verify this.